sdk/vaa/structs.go

```diff
@@ -123,6 +123,17 @@ func (a Address) MarshalJSON() ([]byte, error) {
 	return []byte(fmt.Sprintf(`"%s"`, a)), nil
 }
 
+// Standard marshal stores the Address like this: "[0,0,0,0,0,0,0,0,0,0,0,0,2,144,251,22,114,8,175,69,91,177,55,120,1,99,183,183,169,161,12,22]"
+// The above MarshalJSON stores it like this (66 bytes): ""0000000000000000000000000290fb167208af455bb137780163b7b7a9a10c16""
+func (a *Address) UnmarshalJSON(data []byte) error {
+	addr, err := StringToAddress("0x" + string(data[1:65]))
```

Comment:
If we're going to sub-slice the data then we should do a length check first.  Otherwise we could use `strings.Trim` to trim out the quotes.  Also, the "0x" prefix is optional for `StringToAddress`.

```suggestion
	addr, err := StringToAddress(strings.Trim(data, `"`))
```

Reply:
Fixed.